arangod/Aql/ModificationExecutorHelpers.cpp

```diff
@@ -0,0 +1,224 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2019 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Markus Pfeiffer
+////////////////////////////////////////////////////////////////////////////////
+
+#include "ModificationExecutorHelpers.h"
+
+#include "Aql/AqlValue.h"
+#include "Aql/ModificationExecutorInfos.h"
+#include "Basics/Result.h"
+#include "Utils/CollectionNameResolver.h"
+#include "Utils/OperationResult.h"
+
+#include <velocypack/Builder.h>
+#include <velocypack/Iterator.h>
+#include <velocypack/Slice.h>
+#include <velocypack/velocypack-aliases.h>
+
+#include <string>
+
+#include "Logger/LogMacros.h"
+
+using namespace arangodb;
+using namespace arangodb::aql;
+using namespace arangodb::basics;
+
+Result ModificationExecutorHelpers::getKey(CollectionNameResolver const& resolver,
+                                           AqlValue const& value, std::string& key) {
+  TRI_ASSERT(key.empty());
+
+  // If `value` is a string, this is our _key entry, so we use that.
+  if (value.isString()) {
+    key.assign(value.slice().copyString());
```

Comment:
Could be string_view soon https://github.com/arangodb/arangodb/pull/10287 -- no action needed

Reply:
Already perfect as `assign` takes `&&`